### PR TITLE
Bug 2021446: Set canary status as unknown if not admitted to default ingress controller

### PR DESCRIPTION
**Pull request by gcs278**
Adds a new status for IngressControllerCanaryCheckSuccessConditionType as unknown if the canary route is not admitted to the default ingress controller. It also doesn't attempt to probe canary route if probe routine is already started and canary route is not admitted.

### CODE REVIEW

**Comment by melvinjoseph86** [comment]
Verified via pre-merge verification workflow, more references related to the test can be found in:
https://bugzilla.redhat.com/show_bug.cgi?id=2021446#c8
/label qe-approved

File: pkg/operator/controller/canary/controller.go [hunk 1]
```diff
@@ -244,6 +241,14 @@ func (r *reconciler) startCanaryRoutePolling(stop <-chan struct{}) error {
 			return
 		}
 
+		// Don't attempt to probe if route is not actually admitted.
+		if !checkRouteAdmitted(route) {
+			if err := r.setCanaryNotAdmittedStatusCondition(); err != nil {
+				log.Error(err, "error updating canary status condition")
+			}
+			return
+		}
```
Comment (Miciah) [inline pkg/operator/controller/canary/controller.go:250]: This raises the question, Should we also set the status to "Unknown" in the case that the route doesn't exist?  We can leave that as a follow-up if you prefer.

Comment (gcs278) [reply pkg/operator/controller/canary/controller.go:250]: Even if you go unmanaged, doesn't the ingress controller still ensure the canary route? Is there a situation in which a user can delete the canary route and it not come back (at least try to come back)? I suppose you can never be too sure...

**Comment by Miciah** [comment]
Hunky dory!
/lgtm

**Pull request merged**